### PR TITLE
Schema details

### LINKED ISSUE #237: Ensure Types match what is in data model
Check that the variable types in the Excel Data model correspond to what's in the JSON schema

### LINKED ISSUE #238: Check validation patterns for string fields

### LINKED ISSUE #239: Check enum values for enum fields

### LINKED ISSUE #240: Ensure JSON Schema Fields have meaningful descriptions
Ensure that objects in the JSON schema have meaningful descriptions

### LINKED ISSUE #241: Ensure required fields in JSON schema match data model
Relates to #232

### PR DESCRIPTION
**Pull request by AlexAxthelm**
multi-issue PR to address open details for JSON schema

Builds off #251 
Closes #237 (types)
Closes #238 (regex & length)
Closes #239 (enums)
Closes #241 (required)
Closes #240 (descriptions)

### CODE REVIEW

File: pbtar_schema.json [hunk 1]
```diff
@@ -265,9 +202,7 @@
       "id",
       "name",
       "description",
-      "pathway_type",
-      "pathway_type_tooltip",
-      "modelYearEnd",
+      "pathwayType",
```
[the remark this answers is not in the document]
Comment (AlexAxthelm) [reply pbtar_schema.json:205]: @jdhoffa @jacobvjk Can you confirm that `modelYearStart` and `modelYearEnd` are not mandatory?

[the remark this answers is not in the document]
Comment (jdhoffa) [reply pbtar_schema.json:205]: Agree with Copilot here, `modelYearEnd` is currently used pretty extensively in the website, especially in any "target year" filtering: 
Screenshot 2025-08-13 at 10 48 58

I leave it mainly to @lloyd-rmi and @jacobvjk on how we want to handle this. We can remove the ability to filter by the target year of the scenario, or we can make the field mandatory. 
(Or, alternatively, I suppose we could make it nullable, but then if anyone filters for a specific Target Year, nothing would appear). 

@AlexAxthelm thoughts?

**Review by jdhoffa — COMMENTED** [review]
Mostly looks good to me, pending resolution of the open comment.

File: pbtar_schema.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlexAxthelm) [reply pbtar_schema.json:205]: We should probably add some tests, but my suspicion is that it's fine technically to omit `modelYearStart`, `modelYearEnd`, and `modelYearNetzero`. They just would never show up if someone filters on that field (`Target Year == 2040 ` or `Target Year < 2051`) but should appear on the other filters (`Temperature < 2`).

Overall, I'd say merge as-is, and there will be a second round to address changes once we see this schema in practice.

[the remark this answers is not in the document]
Comment (jdhoffa) [reply pbtar_schema.json:205]: I have a call with Jacob and Lloyd to discuss how we want to address this in the long-term (and agree, we don't need to decide that now). 

However, can you please add 2 tests for the cases you just identified? If we want to merge this as-is, then I want to really be sure that it functions as expected in those edge cases. 

e.g. A scenario with no `modelYearEnd` still appears when the `Temperature == 1.5` filter is selected, but does not appear when the `Target Year == 2040` 

Otherwise I fear we will forget this edge-case

[the remark this answers is not in the document]
Comment (AlexAxthelm) [reply pbtar_schema.json:205]: I've added a minimal test scenario to the `src/data` directory, so that we can visually inspect it. There's some weirdness visually (empty badge) when theres nothing in the "targets" section on the card, and the filter dropdowns now show "undefined", but that isn't a real filter (same as not setting it). but on the other hand, the scenario is showing up in the filters for which it can be filtered.

TL;DR: I wouldn't say that omitting these fields from a scenario file _breaks_ anything, but if we're going to allow them to be unset, we need to handle unset values better.

I think that this should be a follow-up in a separate ticket to make a decision and implement.

Screenshot_2025-08-14_08 36 09

[the remark this answers is not in the document]
Comment (jdhoffa) [reply pbtar_schema.json:205]: Thanks for looking into this Alex. 

> I think that this should be a follow-up in a separate ticket to make a decision and implement.

Agree. Lloyd, Jacob and I will discuss this today, and can get a decision, not necessary to block this PR on that decision. @AlexAxthelm would you mind making that separate ticket so that we follow-up?

**Review by jdhoffa — APPROVED** [review]
Looks great, thank you!

File: pbtar_schema.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlexAxthelm) [reply pbtar_schema.json:205]: #255

[the remark this answers is not in the document]
Comment (jdhoffa) [reply pbtar_schema.json:205]: Thank you!
